xwiki-platform-core/xwiki-platform-rendering/xwiki-platform-rendering-macros/xwiki-platform-rendering-macro-include/src/main/java/org/xwiki/rendering/internal/macro/include/IncludeMacro.java

```diff
@@ -202,7 +204,38 @@ public void setDocumentDisplayer(DocumentDisplayer documentDisplayer)
                 references.pop();
             }
         }
+        
+        if (parameters.getIndentHeadingLevel() > 0) {
+
+            HeaderBlock heading = result.getFirstBlock(new ClassBlockMatcher(HeaderBlock.class), Block.Axes.DESCENDANT);
+
+            if (heading != null) {
+
+                // Define header Levels
+                HeaderLevel[] headerLevels = { HeaderLevel.LEVEL1, HeaderLevel.LEVEL2, HeaderLevel.LEVEL3,
+                        HeaderLevel.LEVEL4, HeaderLevel.LEVEL5, HeaderLevel.LEVEL6 };
+                HeaderLevel headinglevel;
+
+                // Get specified header level
+                if (parameters.getIndentHeadingLevel() > 6) {
+                    headinglevel = headerLevels[5];
+                } else {
+                    headinglevel = headerLevels[parameters.getIndentHeadingLevel() - 1];
+                }
 
+                HeaderBlock newHeading = new HeaderBlock(heading.getChildren(), headinglevel);
+
+                // Get the parent block in document
+                Block parent = result.getChildren().get(0);
+
+                // Get the first child Block of the document
+                Block child = parent.getChildren().get(0);
+                if (child == heading) {
+                    parent.replaceChild(newHeading, child);
```

Comment:
This only seems to replace the first heading.
As stated in the issue:
> When the include macro is inside a SectionBlock it should increment any included header so that included level 1 headers become sub headers of that section.

The possible interpretation of this would be.
1. To have all the level 1 headers be reduced to level 2 (or the specified indent)
1. To have all the headers (level 1 to level 5) be reduced to the specified indent

Reply:
Also, I think you need to check if the include macro is indeed inside a `SectionBlock`.
`SectionBlock` is a section for a `HeadingBlock`. The section which is covered by the heading.

Reply:
Thanks, @9inpachi  for your opinions. I discussed with @vmassol on the issue https://jira.xwiki.org/browse/XWIKI-6721 he advised me to do it only for the first block. Might be he advised just for `remove` let me confirm from him and then I will make the changes.

Reply:
Thanks, @9inpachi  I have updated it.